server/src/framework.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ConnInfo,
+  Handler,
+} from "https://deno.land/std@0.182.0/http/server.ts";
+
+type MethodHandler = { POST?: Handler; GET?: Handler };
+
+interface Router {
+  [index: string]: MethodHandler;
+}
+
+export const app = (router: Router): Handler => {
+  const routes: Array<[RegExp, MethodHandler]> = Object.entries(router).map(
+    ([url_pattern, handler]) => [new RegExp(`^${url_pattern}$`), handler]
```

Comment:
i would factor the inner function to globabl scope, and inline `routes`

then you can drop the curly braces and the whole function body would be a one liner